plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/rest/TrinoIcebergRestCatalogFactory.java

```diff
@@ -71,6 +74,7 @@ public synchronized TrinoCatalog create(ConnectorIdentity identity)
         if (icebergCatalog == null) {
             ImmutableMap.Builder<String, String> properties = ImmutableMap.builder();
             properties.put(CatalogProperties.URI, serverUri.toString());
+            warehouse.ifPresent(location -> properties.put(CatalogProperties.WAREHOUSE_LOCATION, location));
```

Comment:
@danielcweeks  I set random characters with `iceberg.rest-catalog.warehouse` in `IcebergRestQueryRunnerMain`, but I didn't find any change. I expected the REST catalog throws an error or log something because the behind JDBC catalogs uses the different `warehouse` catalog property. Can you share how REST catalog behaves in this case?

Reply:
@ebyhr this is how the REST catalog currently uses this property: https://github.com/apache/iceberg/blob/4b5ee1a0ce77161b9195d8db4fe0e8350ea93fa8/core/src/main/java/org/apache/iceberg/rest/RESTSessionCatalog.java#L736-L746.

Reply:
Thanks, but the link doesn't answer the above question in my opinion.

Reply:
@ebyhr I tried to explain in [this comment](https://github.com/trinodb/trino/pull/15762#discussion_r1082839039), but this can't really be compared to how Hive works with locations and the expectation is not that the value should necessarily dictate anything about the path.

Because Iceberg has multiple implementations for Catalog, all with different behaviors, the `warehouse` property is just a standard mechanism by which to indicate intent to the implementation.  For simple implementations, like Hadoop or Hive Catalog, that could be just a physical location to store data.  However, it is more of an abstraction that can be used by the implementation.

For example, the catalog service at Netflix has multiple catalogs (`prod`, `test`, `shadow`, etc.), but there's just a single service endpoint (say `metacat.data.netflix.net`).  To address each of those independently, you need to either have different domains for each or expose different ports to the same service.  With RESTCatalog implementation, you're just providing a request to address a specific one without having to change the surrounding infrastructure (and something like `prod` is a lot easier to reason about than port `337033`).

The RESTSpect defines this as an option, but does not dictate how it is handled (and ignoring it is also an option, since that is a catalog behavior).

We could write tests, but at that point we'd be writing both the test and the expected behavior, so the test would largely be meaningless.